### PR TITLE
applications: nrf_desktop: Integrate APPLICATION_CONFIG_DIR for sysbuild

### PR DESCRIPTION
**Pull request by mkapala-nordic**
```
Change integrates APPLICATION_CONFIG_DIR for sysbuild. Change also fixes
a few configuration issues observed while testing locally.

Jira: NCSDK-24857
Jira: NCSDK-27858
Jira: NCSDK-27568
```
and
```
applications: nrf_desktop: Remove bootloader-related ops from app confs

After migration to sysbuild, now bootloader-specific Kconfig options
are driven at the sysbuild level, not the main application.
Remove unecessary options that were moved to proper sysbuild*.conf
files.

Jira: NCSDK-27890
```

~Depends on:  https://github.com/nrfconnect/sdk-nrf/pull/15530~

Review only last two commits.

### CODE REVIEW

File: applications/nrf_desktop/Kconfig.sysbuild
Comment (MarekPieta) [inline applications/nrf_desktop/Kconfig.sysbuild:1]: Should you appear under both `Co-authored-by` and `Signed-off-by` in the commit message?

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/sysbuild.conf
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840dk_nrf52840/sysbuild.conf:9]: Do we still need to set the `CONFIG_SB_SIGNING_KEY_FILE` in application configuration? Similar for other Kconfig options for B0 bootloader.

File: applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/sysbuild_mcuboot_smp.conf
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/sysbuild_mcuboot_smp.conf:10]: It used to be `CONFIG_BOOT_SIGNATURE_TYPE_ECDSA_P256` here

File: applications/nrf_desktop/configuration/nrf5340dk_nrf5340_cpuapp/sysbuild.conf
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf5340dk_nrf5340_cpuapp/sysbuild.conf:12]: Remove the empty line, please

File: applications/nrf_desktop/sample.yaml
Comment (MarekPieta) [inline applications/nrf_desktop/sample.yaml:178]: Can we merge entries for `nrf54h20` with remaining targets?

Comment (MarekPieta) [reply applications/nrf_desktop/sample.yaml:178]: Special handling should no longer be needed.

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/sysbuild_mcuboot_smp.conf
Comment (mkapala-nordic) [reply applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/sysbuild_mcuboot_smp.conf:10]: Oh, it is an only config that has it set, must have missed that. Thanks

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/sysbuild.conf
Comment (mkapala-nordic) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/sysbuild.conf:9]: It is separate ticket that was not addressed yet here -> [NCSDK-27890](https://nordicsemi.atlassian.net/browse/NCSDK-27890)

File: applications/nrf_desktop/Kconfig.sysbuild
Comment (mkapala-nordic) [reply applications/nrf_desktop/Kconfig.sysbuild:1]: `Co-authored-by` is GitHub feature that shows that this commit has multiple authors and `Signed-off-by` is a requirement so the compliance passes.
Probably I could remove `Signed-off-by` and still satisfy compliance but as is also should work.
We will check it after I'll remove the draft status.

Comment (MarekPieta) [reply applications/nrf_desktop/Kconfig.sysbuild:1]: Thanks for clarification

**Review by MarekPieta — APPROVED** [review]
nRF Desktop changes looks good

**Comment by shanthanordic** [comment]
Looks like big change after RC-1 @pdunaj  @kapi-no  Can I push it out of NCs 2.7.0 unless absolutely needed

**Comment by kapi-no** [comment]
> Looks like big change after RC-1 @pdunaj @kapi-no Can I push it out of NCs 2.7.0 unless absolutely needed

If we want to maintain the production quality of the nRF Desktop app, we need this change that is blocked on the @tejlmand PR (dependency indicated in the PR description). Alternatively, we can lower the nRF Desktop quality from production to experimental following the VS code approach. However, going back to the experimental quality for NCS 2.7.0 needs wider discussion from relevant stakeholders.

On a separate note, we had a different arrangement with @carlescufi and @tejlmand, as this change was supposed to be part of NCS 2.7.0.

**Review by tejlmand — APPROVED** [review]

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/prj.conf
Comment (mkapala-nordic) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/prj.conf]: removed sections and merged with zephyr configs, comment is not necessary imo

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/prj_mcuboot_qspi.conf
Comment (mkapala-nordic) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/prj_mcuboot_qspi.conf]: removed section and merged with zephyr configs

Comment (mkapala-nordic) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/prj_mcuboot_qspi.conf]: Follow-up to verify if they are needed: https://nordicsemi.atlassian.net/browse/NCSDK-27950

File: applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/prj_mcuboot_smp.conf
Comment (mkapala-nordic) [reply applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/prj_mcuboot_smp.conf]: Follow-up to verify if they are needed: https://nordicsemi.atlassian.net/browse/NCSDK-27950

**Review by MarekPieta — APPROVED** [review]

**Comment by mkapala-nordic** [comment]
Rebased only

**Comment by mkapala-nordic** [comment]
Rebased

**Comment by mkapala-nordic** [comment]
Removed `CONFIG_BOOT_DIRECT_XIP=y` from MCUboot configs as it is propagated from sysbuild (`nrf/sysbuild/CMakeLists.txt:L228`).

**Review by zycz — APPROVED** [review]

**Comment by anangl** [comment]
Rebased after #15530 got merged.

**Review by kapi-no — APPROVED** [review]
Approving it due to imminent RC2.

**Pull request merged**
